front/src/applications/operationalStudies/hooks/useSelectedTrainAndPathForProjection.ts

```diff
@@ -0,0 +1,47 @@
+import type { InfraWithState } from 'common/api/osrdEditoastApi';
+import { useEffect } from 'react';
+import { useAppDispatch } from 'store';
+
+import {
+  updateSelectedTrainId,
+  updateTrainIdUsedForProjection,
+} from 'reducers/osrdsimulation/actions';
+import { useSelector } from 'react-redux';
+import { getSelectedTrainId, getTrainIdUsedForProjection } from 'reducers/osrdsimulation/selectors';
+import type { TrainScheduleWithDetails } from 'modules/trainschedule/components/Timetable/types';
+
+const useSelectedTrainAndPathForProjection = (
+  infra: InfraWithState,
+  trainIds: number[],
+  trainScheduleSummaries: TrainScheduleWithDetails[]
+) => {
+  const dispatch = useAppDispatch();
+  const currentProjection = useSelector(getTrainIdUsedForProjection);
+  const selectedTrainId = useSelector(getSelectedTrainId);
+
+  useEffect(() => {
+    if (infra.state === 'CACHED' && trainIds.length > 0 && !currentProjection) {
+
+      // if a selected train is given, we use it for the projection
+      if (selectedTrainId && !currentProjection && trainIds.includes(selectedTrainId)) {
+        dispatch(updateTrainIdUsedForProjection(selectedTrainId));
+        return;
+      }
+
+      // if there is already a projection and the projected train still exists, keep it
+      if (currentProjection && trainIds.includes(currentProjection)) {
```

Comment:
Is it possible to go in this condition since we check at first for `!currentProjection` ?

Reply:
Actually, the first condition !currentProjection l23 was not correct, I removed it